kobo/apps/reports/serializers.py

```diff
@@ -24,7 +25,11 @@ def to_representation(self, obj):
 
         split_by = request.query_params.get('split_by', None)
 
-        _list = report_data.data_by_identifiers(obj, vnames, split_by=split_by)
+        permission_filters = obj.get_filters_for_partial_perm(request.user.id)
+        submission_stream = obj.deployment.get_submissions(
```

Comment:
I'm wondering if we should add a required parameter called something like `requesting_user` to `get_submissions()` and then have that method handle filtering for partial permissions. The goal would be to have the filtering logic in one place and to break (intentionally) any code that's retrieving submissions without making sure the requestor is entitled to see the full set.

Reply:
Indeed.